internal/v1/handler.go

```diff
@@ -229,27 +230,40 @@ func (h *Handlers) GetComposeStatus(ctx echo.Context, composeId uuid.UUID) error
 		return err
 	}
 
-	resp, err := h.server.cClient.ComposeStatus(composeId)
+	var resp *http.Response
+	var body []byte
+
+	err = retry.Do(
+		func() error {
+			var err error
+			resp, err = h.server.cClient.ComposeStatus(composeId)
+			if err != nil {
+				return err
+			}
+			defer closeBody(ctx, resp.Body)
+
+			body, err = io.ReadAll(resp.Body)
+			return err
+		},
+		retry.Attempts(3),
+		retry.Delay(time.Second),
+		retry.DelayType(retry.BackOffDelay),
+		retry.OnRetry(func(n uint, err error) {
+			ctx.Logger().Warnf("Attempt %d failed: %v. Retrying...", n+1, err)
```

Comment:
This log probably needs some more context, maybe incude the composeId here too (or the url)